packages/lib/zuauth/src/server.ts

```diff
@@ -31,14 +33,15 @@ export async function authenticate(
   const pcd = await ZKEdDSAEventTicketPCDPackage.deserialize(serializedPCD.pcd);
 
   if (!(await ZKEdDSAEventTicketPCDPackage.verify(pcd))) {
-    throw new Error("ZK ticket PCD is not valid");
+    throw new ZuAuthAuthenticationError("ZK ticket PCD is not valid");
   }
 
   if (pcd.claim.watermark.toString() !== watermark) {
```

Comment:
I think the externalNullifier should be passed in by the caller to be verified here too (if it's set at all).
In general, it should be verified by anyone who makes use of the nullifierHash.

Reply:
The external nullifier seems redundant if the watermark is used. What extra security or utility is it providing?